src/utils.jl

```diff
@@ -4,3 +4,26 @@ function sum_if_not_none_or_initialize(A,B)
     end
     return A + B
 end
+
+function contiguous_regions(condition::AbstractVector{Bool})
+    # Find the indicies of changes in "condition"
+    d = diff(condition)
+    idx = findall(x->x != 0, d)
+
+    # We need to start things after the change in "condition". Therefore, 
+    # we'll shift the index by 1 to the right.
+    idx .+= 1
+
+    if condition[1]
+        # If the start of condition is True prepend a 0
+        idx = append!([1],idx)
+    end
+
+    if condition[end]
+        # If the end of condition is True, append the length of the array
+        idx = append!(idx,[condition.size+1]) # Edit
+    end
+
+    # Reshape the result into two columns
+    return reshape(idx,2,length(idx) ÷ 2)'
```

Comment:
What happens if the length of `idx` isn't even?

Reply:
Length of idx will always be even as if there is 010 or 101 in the body (excluding the ends) it diff is nonzero at two places. Else it start or end will be 1 and thus they are added .